java/src/main/java/com/microsoft/tunnels/connections/TcpipForwardRequestHandler.java

```diff
@@ -55,14 +55,28 @@ public Result process(
         connectionService.getForwarder(), "No TCP/IP forwarder");
 
     String address = buffer.getString();
-    int requested = buffer.getInt();
-    int port = requested;
+    int requestedPort = buffer.getInt();
+
+    // Ignore the request if the port is already in the collection.
```

Comment:
I thought this shouldn't ever happen if the host is operating correctly. I'd like to understand why it is happening, before ignoring it.

Reply:
(As I recall, the client was _not_ making current requests, so I don't think that is the explanation.)

Reply:
I've updated the PR to only add refresh ports tests.